appservice/src/createAppService/LogAnalyticsCreateStep.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtLocation, getResourceGroupFromId, LocationListStep, uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { AzureWizardExecuteStep, nonNullProp, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import { Progress } from "vscode";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { createOperationalInsightsManagementClient } from "../utils/azureClients";
+import { AppInsightsCreateStep } from "./AppInsightsCreateStep";
+import { IAppServiceWizardContext } from "./IAppServiceWizardContext";
+
+export class LogAnalyticsCreateStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 134;
+
+    public async execute(context: IAppServiceWizardContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        const opClient = await createOperationalInsightsManagementClient(context);
+        const rgName = nonNullValueAndProp(context.resourceGroup, 'name');
+        const resourceLocation: AzExtLocation = await LocationListStep.getLocation(context);
+        const aiStep = new AppInsightsCreateStep();
```

Comment:
This is a strange pattern. Execute steps shouldn't be creating other execute steps...seems like we just need a utility method for `getSupportedLocation`

Reply:
Yeah, I did this because it's the easiest way to get access to that location property without a huge refactor since that function uses a lot of private functions that exist on the class.  Also, a util function seems a bit overkill since this is specifically for App Insights and nothing else would be utilizing it.

I had another idea to get this information though so I'll do that instead.

Reply:
Never mind, the refactor wasn't  hard at all and all my other ideas had some other weird issues, so just moved it to its own file 👍